src/controller/CHIPDeviceController.cpp

```diff
@@ -1889,12 +1891,23 @@ void DeviceCommissioner::OnScanNetworksResponse(void * context,
                                                : "none provided"));
     DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);
 
+    // advance to the kNeedsNetworkCreds waiting step
+    commissioner->CommissioningStageComplete(CHIP_NO_ERROR);
+
```

Comment:
It really seems here that we want to SET the state to `kNeedsNetworkCreds`, not "advance" (which could break the state machine if there were more states added in between, where the next step will be advanced by the delegate.